src/lib/openjp2/t1.c

```diff
@@ -1786,6 +1786,34 @@ static void opj_t1_clbl_decode_processor(void* user_data, opj_tls_t* tls)
                 }
             }
 #endif
+
+#ifdef __loongarch_asx
+            {
+                asm volatile
```

Comment:
Using "volatile" is generally a bad idea (unless there is a side-effect not explainable by the input/output/clobber).  If it does not work w/o "volatile", you've likely missed some register input/output/clobber.

Reply:
affect performance or something else ?

Reply:
Because "if it's a chicken, don't model it as a duck".

Reply:
Please , provide helpful suggestions.

Reply:
Is it broken after if we remove `volatile`?  If it's not broken, just remove `volatile`.  Otherwise try to figure out why the compiler breaks it and fix the input/output/clobber list.

`volatile` **means** there is some unexplainable side effect, for example writing/reading some CSRs, invoking hardware memory barriers, etc.  These just do not apply for a SIMD optimization.

Reply:
just want to tell compiler not to optimize or modify the asm block.

Reply:
> just want to tell compiler not to optimize or modify the asm block.

It won't modify the asm block because the compiler does not parse asm at all.

The only possible optimizations are:

1. Reorder the asm block with the instructions generated by the compiler itself.
2. Remove the asm block completely.

If the input/output/clobber lists are correct, 2 should be disabled, and 1 should be performed in a constrained way not to break the code.

Reply:
Well, and maybe

3. Duplicating the asm block due to a loop unrolling etc.  Again if the input/output/clobber lists are correct this should happen in a safe way.